### ISSUE TITLE
Handle SPIR-V local structures without offset decorations.

### ISSUE THREAD

**Issue by kvark** [gfx-rs/naga#974]
Trace:  [wgpu_trace.zip](https://github.com/gfx-rs/naga/files/6658842/wgpu_trace.zip)
Error:
> Type [24] 'ShadowDirectionalLight' is invalid:
	Structure member[0] at 0 and size 64 crosses the structure boundary of size 16

The SPIR-V module has the same structure defined twice: once with offsets, and once without offsets:
```
               OpName %ShadowDirectionalLight "ShadowDirectionalLight"
               OpMemberName %ShadowDirectionalLight 0 "viewProj"
               OpMemberName %ShadowDirectionalLight 1 "pos"
               OpMemberName %ShadowDirectionalLight 2 "shadow_bias_min_max"
               OpMemberName %ShadowDirectionalLight 3 "textureIndex"
               OpMemberName %ShadowDirectionalLight 4 "_padding"
               OpName %shadow_light "shadow_light"
               OpName %ShadowDirectionalLight_0 "ShadowDirectionalLight"
               OpMemberName %ShadowDirectionalLight_0 0 "viewProj"
               OpMemberName %ShadowDirectionalLight_0 1 "pos"
               OpMemberName %ShadowDirectionalLight_0 2 "shadow_bias_min_max"
               OpMemberName %ShadowDirectionalLight_0 3 "textureIndex"
               OpMemberName %ShadowDirectionalLight_0 4 "_padding"
```

Our validator doesn't like the second one.

**Comment by superdump** [comment]
The `shadow_light` is from having a variable inside a for loop like `ShadowDirectionalLight shadow_light = shadow_directional_lights[i];` and then using `shadow_light` for the rest of the loop body. As @kvark noted, removing that and using `shadow_directional_lights` directly allowed the code to work.

**Comment by JCapucho** [comment]
Fixed by #1326

The validator trips up on `data12.spv` because of texture sampling on non uniform control flow but that's expected
